data-products/src/main/scala/org/sunbird/analytics/job/report/StateAdminReportJob.scala

```diff
@@ -400,10 +400,11 @@ object StateAdminReportJob extends optional.Application with IJob with StateAdmi
   }
     
     def locationIdListFunction(location: String): List[String] = {
+        val optLocation = Option(location)
         var locations = new ListBuffer[String]()
         try {
-            if (location != null) {
-                val profileLocation = JSONUtils.deserialize[List[Map[String, String]]](location)
+            if (optLocation != null && !optLocation.isEmpty) {
```

Comment:
no need of null check and check for optLocation.isDefined rather than negation on empty

Reply:
@SanthoshVasabhaktula done the change